File: Algorithms/Implementation/Sherlock-And-Squares.cc
// Author : Sunil BN
// Copyright : Copyright 2015, hacker_rank Project
// License : MIT
// Version : 1.0.0
// Email : sunhick@gmail.com

#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

int count_squares(int a, int b) {
  int counter;
  // Intermediate value theorem
  // https://www.mathsisfun.com/algebra/intermediate-value-theorem.html
  // https://en.wikipedia.org/wiki/Intermediate_value_theorem
  int sqrta = (int)(ceil(sqrt(a)));
  int sqrtb = (int)(floor(sqrt(b)));
  counter = sqrtb - sqrta + 1;     
  return counter;
}

int main() {
  int testcases;
  cin >> testcases;

  int a, b;
  for (int i = testcases; i > 0; i--) {
    cin >> a >> b;
    cout << count_squares(a, b) << endl;
  }

  return 0;
}

File: Algorithms/Implementation/Sherlock-And-Squares.py
#!/bin/python3

import math

__author__ = "Sunil"
__copyright__ = "Copyright 2015, hacker_rank Project"
__license__ = "MIT"
__version__ = "1.0.0"
__email__ = "sunhick@gmail.com"


def count_squares(start, end):
    return int(math.floor(math.sqrt(end)) - math.ceil(math.sqrt(start)) + 1)

if __name__ == '__main__':
    testcases = int(input().strip())

    for i in range(0, testcases):
        start, end = input().strip().split(' ')
        print(count_squares(int(start), int(end)))
